compiler/src/iree/compiler/Codegen/Common/TileDispatchUsingForall.cpp

```diff
@@ -516,6 +538,20 @@ void TileAndDistributeToWorkgroupsUsingForallOpPass::runOnOperation() {
       // TODO: run producer and consumer fusion in one worklist.
       fuseProducersOfSlices(rewriter, newFusionOpportunities,
                             tileAndFuseOptions, newLoop);
+      forallOp = newLoop;
+    }
+
+    // Reorder the workgroups if the strategy is set to `transpose`.
+    // This just transposes the first two dimensions of the workgroup i.e., the
+    // #iree.codegen.workgroup_id_x and #iree.codegen.workgroup_id_y.
+    // Only reorders if the loop bounds are static.
+    if (transposeWorkgroup) {
+      SmallVector<Attribute> mappingAttrs(forallOp.getMappingAttr().getValue());
```

Comment:
I'm confused.. How does it work? I thought that the `forallOp` is declared in the above block. Should it be `tilingLoops[0]`?

Reply:
It's declared here (in the same block) https://github.com/iree-org/iree/blob/5ee9b27635e99df5edeb52bbb46cafb224b03a7d/compiler/src/iree/compiler/Codegen/Common/TileDispatchUsingForall.cpp#L503 but again updated here: https://github.com/iree-org/iree/blob/5ee9b27635e99df5edeb52bbb46cafb224b03a7d/compiler/src/iree/compiler/Codegen/Common/TileDispatchUsingForall.cpp#L510 I am just recapturing it in the if block.

Reply:
Oh, they are in the same block. I can't count the number of indent.